src/main/java/application/rest/v1/configmaps/KindActionMappingProcessor.java

```diff
@@ -0,0 +1,462 @@
+            if ((subkind != null) && !subkind.isEmpty()) {
+                if ((name != null) && !name.isEmpty()) {
+                    retVal = KSN;
+                } else {
+                    retVal = KS;
+                }
+            } else if ((name != null) && !name.isEmpty()) {
+                retVal = KN;
+            } else {
+                retVal = K;
+            }
+        } else {
+            if (Logger.isErrorEnabled()) 
+                Logger.log(className, "exemineMappingProperties", Logger.LogType.ERROR, 
+                           "kind given is null or empty string");
+        }
+
+        if (Logger.isExitEnabled()) 
+                Logger.log(className, "exemineMappingProperties", Logger.LogType.EXIT, 
+                           "retVal = " + retVal);
+
+        return retVal;
+    }
+
+    /**
+     * Find a matched configmap defined in a given KAM 
+     * 
+     * @param cNumFields
+     * @param kNumFields
+     * @param cName
+     * @param cSubkind
+     * @param cKind
+     * @param gName
+     * @param gSubkind
+     * @param gKind
+     * @return true if found; otherwise false
+     */
+    private boolean foundMatchedConfigMap(int cNumFields, int kNumFields,
+                              String cName, String cSubkind, String cKind,
+                              String gName, String gSubkind, String gKind) {
+
+        if (Logger.isEntryEnabled()) 
+                Logger.log(className, "foundMatchedConfigMap", Logger.LogType.ENTRY, 
+                           "cNumField = " + cNumFields + " kNumField = " + kNumFields);
+
+        if ( (cNumFields == -1) || (kNumFields == -1)) {
+            if (Logger.isExitEnabled()) 
+                Logger.log(className, "isMatched", Logger.LogType.EXIT, "return false");
+            return false;
+        }
+
+        boolean match = false;
+        if (cNumFields == kNumFields) {
+            switch (kNumFields) {
+            case KSN:
+                if ((gName.equals(cName)) || (gName.equals(WILDCARD)))
+                    if ((gSubkind.equals(cSubkind)) || (gSubkind.equals(WILDCARD)))
+                        if ((gKind.equals(cKind)) || (gKind.equals(WILDCARD)))
+                            match = true;
+                break;
+            case KS:
+                if ((gSubkind.equals(cSubkind)) || (gSubkind.equals(WILDCARD)))
+                    if ((gKind.equals(cKind)) || (gKind.equals(WILDCARD)))
+                        match = true;
+                break;
+            case KN:
+                if ((gName.equals(cName)) || (gName.equals(WILDCARD)))
+                    if ((gKind.equals(cKind)) || (gKind.equals(WILDCARD)))
+                        match = true;
+                break;
+            case K:
+                if (cNumFields == K)
+                    if ((gKind.equals(cKind)) || (gKind.equals(WILDCARD)))
+                        match = true;
+            }
+        } else if (kNumFields == K) {
```

Comment:
Suggestion: might want to switch order of test done on 381 with the test done on 384.  I know you get the same outcome either way.  It's just that if you change the order,  the order of condition in this method would match the hierarchy order - i.e. KSN, KS, K - and therefore be more intuitive to the reader and raise less questions.  After I saw the order of the condition tests in this method of 'KSN, K, KS' it made me concerned the order was wrong and I had to study the caller of this method to determine the order of testing the conditions doesn't actually matter.